### PR TITLE
feat: add gav matching and overwritable mediatype for file types

### PR DESCRIPTION
**Pull request by Skarlso**
#### Which issue(s) this PR fixes

https://github.com/open-component-model/ocm-project/issues/200

### CODE REVIEW

File: api/ocm/extensions/blobhandler/handlers/generic/maven/blobhandler.go [hunk 1]
```diff
@@ -129,3 +140,53 @@ func (b *artifactHandler) StoreBlob(blob cpi.BlobAccess, resourceType string, hi
 	}
 	return access.New(url, coords.GroupId, coords.ArtifactId, coords.Version), nil
 }
+
+// PomGav defines gav information in a POM file.
+type PomGav struct {
+	GroupID    string `xml:"groupId"`
+	ArtifactId string `xml:"artifactId"`
```
Comment (hilmarf) [inline api/ocm/extensions/blobhandler/handlers/generic/maven/blobhandler.go:147]:
```suggestion
	GroupId    string `xml:"groupId"`
	ArtifactId string `xml:"artifactId"`
```

or is there a special golang naming convention?

Comment (Skarlso) [reply api/ocm/extensions/blobhandler/handlers/generic/maven/blobhandler.go:147]: No, just conform to `ArtifactId` down below... :)

**Review by hilmarf — COMMENTED** [review]
can you rename `GroupID` to `GroupId`?

File: api/ocm/extensions/blobhandler/handlers/generic/maven/blobhandler.go (hunk 1 above)
Comment (Skarlso) [reply api/ocm/extensions/blobhandler/handlers/generic/maven/blobhandler.go:147]: WAIT. I thought I did rename it hahahahaha. So sorry.

**Comment by Skarlso** [comment]
Fixing the test....

**Review by hilmarf — APPROVED** [review]

**Pull request merged**